File: JAVA/590. N-ary Tree Postorder Traversal.java
class Solution {
    List<Integer>n=new ArrayList<>();
    void p(Node root){
        if(root ==null)
        return;
        for(Node c:root.children){
            p(c);
            n.add(c.val);
        }
    }
    public List<Integer> postorder(Node root) {
        if(root==null)
        return n;
        p(root);
        n.add(root.val);
        return n;
    }
}

File: C++/590. N-ary Tree Postorder Traversal.cpp
class Solution {
    vector<int>n={};
    void p(Node* root){
        if(root==NULL)
        return ;
        for(auto c:root->children){
            p(c);
            n.push_back(c->val);
        }
    }
public:
    vector<int> postorder(Node* root) {
        if(root==NULL)
        return n;
        p(root);
        n.push_back(root->val);
        return n;     
    }
};